app/domain/authentication/o_auth/verify_and_decode_token.rb

```diff
@@ -41,7 +46,7 @@ def fetch_certs(force_read: false)
 
         @jwks = provider_certificates.jwks
         @algs = provider_certificates.algorithms
-        @logger.debug(Log::OIDCProviderCertificateFetchedFromCache.new)
+        @logger.debug(Log::IdentityProviderCertificateFetchedFromCache.new)
       end
 
       def verify_and_decode_token
```

Comment:
We have
```
    def verify_and_decode_token
        ensure_certs_are_fresh
        verified_and_decoded_token
      end

      def ensure_certs_are_fresh
        verified_and_decoded_token
```
Why are we calling `verified_and_decoded_token` twice here? From `verify_and_decode_token` and `ensure_certs_are_fresh`

Reply:
Yes. This is how `ensure_certs_are_fresh` works. I'll add a comment in the code:
```
      # ensure_keys_are_fresh will try to verify and decode the token and if it 
      # fails we fetch the provider keys again. We then verify and decode again
      # so we definitely don't fail because the keys are too old. If ensure_keys_are_fresh
      # will succeed to decode the token then the call to verified_and_decoded_token 
      # will not do a thing because the object is memoized
```